src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/PooledObjects/PooledArrayBuilder`1.cs

```diff
@@ -468,13 +466,343 @@ public bool TryPop([MaybeNullWhen(false)] out T item)
     }
 
     /// <summary>
-    ///  This is present to help the JIT inline methods that need to throw
-    ///  a <see cref="IndexOutOfRangeException"/>.
+    ///  Determines whether this builder contains any elements.
+    /// </summary>
+    /// <returns>
+    ///  <see langword="true"/> if this builder contains any elements; otherwise, <see langword="false"/>.
+    /// </returns>
+    public readonly bool Any()
+        => Count > 0;
+
+    /// <summary>
+    ///  Determines whether any element in this builder satisfies a condition.
+    /// </summary>
+    /// <param name="predicate">
+    ///  A function to test each element for a condition.
+    /// </param>
+    /// <returns>
+    ///  <see langword="true"/> if this builder is not empty and at least one of its elements passes
+    ///  the test in the specified predicate; otherwise, <see langword="false"/>.
+    /// </returns>
+    public readonly bool Any(Func<T, bool> predicate)
```

Comment:
Should we add `TArg`-carrying variants now, rather than having to add them later when we see perf problems? Note, I would prefer if we did `Any(TArg, Func)`, rather than `Any(Func, TArg)`, as the former lends itself to a better IDE experience.

Reply:
I agree that the IDE experience is better if `TArg` goes first. Shall I go ahead and swap the order of the existing `Any` and `All` variants in the compiler?

Reply:
Absolutely. I'm slowly working on a PR for Roslyn to do the same :).

Reply:
Sounds good! Happy to make things consistent.